### ISSUE TITLE
Crew objectives (again!)

### ISSUE THREAD

**Issue by Carbonhell** [HippieStation/HippieStationdeprecated2020#8715]
[Title]: Crew objectives

[Implementation]: I did code them in one of our past rebases, and I believe I coded them so that they'd be pretty modularized, so porting them back would be easy. What would be harder is actually making decent objectives, the system was liked i believe, but the objectives themself were a bit shitty. I admit I made up some randomly to get the initial idea in and then just forgot to improve them, so maybe some discussion would greatly help this idea.

[Why]: It'd enhance crew doing their job, for one. Juicy greentext is always good. Plus, now that we do actually have the loadout system, we could also implement a minimum_points and maximum_points system, so that you get more points (capped) if you're a good lad who does his job.

[Resources]: https://github.com/paperstation/archives/blob/c46f9ce61ccf7157e1a6ffa21222857dce45c557/hippiestation-2016/code/game/gamemodes/objective.dm#L938
It probs can be improved, anyway. But honestly i think it's safe to say the code will work if you simply slap it in.

# Be sure to ask some maintainer to label this issue as a feature. Either ping them on our discord or PM them on our forum. Thank you for sharing your ideas!
@HippieStation/maintainers

**Comment by JohnGinnane** [comment]
idk if it's a good idea but we can also reward via antag points

**Comment by ghost** [comment]
maybe the thing tg added where playing normal rounds makes you more likely to be a traitor? but modified so actually completing your job objectives increses the chance?

**Comment by ghost** [comment]
make them random so every job has a pool of a different ones

**Comment by ghost** [comment]
yea cool +1

**Comment by MiniMeatwad** [comment]
highly recommend this +1

**Comment by Carbonhell** [comment]
This feature needs possible objectives or the +1 is pointless, someone who can think of good objectives is needed

**Comment by ghost** [comment]
Janitor:
Wash 130 cleanable decals
Never dirty the floor with sprays or crayons
End the round having your trashbag full of dead mice/glass shards
slip people up 3x(player count) times
the shuttle must be impeccable 
the janitors lizard must survive and escape alive on the emergency shuttle
no mice in maint
never take off your galoshes
never take off your biosuit

**Comment by ghost** [comment]
Mime
never break your vow (adds quite a few loadout points for next round) 
let there be silence (kill every single department pet on the station) 
backstab using a butterfly knife 10 times
escape with 0.5x(player count) asses in your inventory,  quell the heretic and loud farts
use your pen to tag 50 tiles of the station,  the clown must not tag 50 tiles of the station (if mime rolls this clown also does but on the other side) 
destroy/despoil the clowns bananium structure 
the mimes cat must escape alive on the emergency shuttle

**Comment by ghost** [comment]
clown
never take your shoes off (this is a default objective and not completing it nulls any loadout points gain.  same with mime breaking vow) 
slip people up 2xplayercount on your pda
escape alive on the emergency shuttle in a honkermech (loadout reward reaches maxcap) 
consume 1000u of banana juice
creampie (number of pies in clown locker) people (not that easy) 
the tag objective mime also has
export bananium to centcomm 
end the round with 10 bananas in your inventory 
honkers must escape alive on the emergency shuttle
the mimes pet must die (mime gets opposite protect and assassination objective) 
you must end the round wielding a musical instrument on the emergency shuttle 
hit people with your honker/airhorn 200 times
successfully activate a clown bombs countdown
become a cluwne

**Comment by Eagle-Eyess** [comment]
+1 while I don’t have the time right now to think of a bunch of objs. I will say that I believe each department should get their own types of objs (like before) and heads would get the same random objs like their departments but also with a protect their pet obj since every head has one now. +1

**Comment by ghost** [comment]
https://github.com/OracleStation/OracleStation/pull/626

**Comment by ghost** [comment]
@Carbonhell 
thoughts

**Comment by Carbonhell** [comment]
Yeah, +1. I want them badly, but we will probably have to discuss in the PR implementation for whatever objective the coder implementing this decides to add. That said, the framework itself is accepted.

**Comment by Jujumatic** [comment]
porting this back should be easyish but likely time consuming, i would recommend about $2-$3 for this
